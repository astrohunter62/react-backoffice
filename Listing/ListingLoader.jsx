import React from 'react'
import PropTypes from 'prop-types'

import { withStyles } from 'material-ui/styles'
import Table, {
  TableCell,
  TableRow,
} from 'material-ui/Table'
import { LinearProgress } from 'material-ui/Progress'

const styles = (theme) => ({
  row: {
    height: 'auto',
  },
  cell: {
    border: 0,
    padding: 0,
  },
  progress: {
    width: '100%',
  },
})

const ListingLoader = ({ cols, classes }) => (
  <TableRow className={classes.row}>
    <TableCell padding="none" colSpan={cols} className={classes.cell}>
      <LinearProgress mode="query" className={classes.progress} />
    </TableCell>
  </TableRow>
)

ListingLoader.propTypes = {
  cols: PropTypes.number.isRequired,
  classes: PropTypes.object.isRequired,
}

ListingLoader.defaultProps = {
  cols: 1,
}

export default withStyles(styles)(ListingLoader)
