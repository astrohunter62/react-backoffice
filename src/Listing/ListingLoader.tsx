import React from "react";
import {
  TableCell,
  TableRow,
  LinearProgress,
  withStyles,
} from "@material-ui/core";

const styles = {
  row: {
    height: "auto",
  },
  cell: {
    border: 0,
    padding: 0,
  },
  progress: {
    width: "100%",
  },
};

type ListingLoaderProps = {
  cols?: number;
  classes: {
    [key: string]: string;
  };
};

const ListingLoader: React.SFC<ListingLoaderProps> = ({ cols, classes }) => (
  <TableRow className={classes.row}>
    <TableCell padding="none" colSpan={cols} className={classes.cell}>
      <LinearProgress variant="query" className={classes.progress} />
    </TableCell>
  </TableRow>
);

ListingLoader.defaultProps = {
  cols: 1,
};

export default withStyles(styles)(ListingLoader);
