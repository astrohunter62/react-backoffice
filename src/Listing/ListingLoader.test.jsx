import React from 'react'
import renderer from 'react-test-renderer'
import { Table } from 'material-ui'

import ListingLoader from './ListingLoader'

it('renders correctly', () => {
  const tree = renderer
    .create((
      <Table>
        <ListingLoader cols={1} />
      </Table>
    ))
    .toJSON()
  expect(tree).toMatchSnapshot()
})
