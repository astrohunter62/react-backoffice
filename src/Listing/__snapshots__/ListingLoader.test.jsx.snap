// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<table
  className="MuiTable-root-1"
>
  <tr
    className="MuiTableRow-root-5 ListingLoader-row-2"
  >
    <td
      aria-sort={null}
      className="MuiTableCell-root-10 ListingLoader-cell-3"
      colSpan={1}
      scope={undefined}
    >
      <div
        className="MuiLinearProgress-root-18 MuiLinearProgress-primaryColor-19 ListingLoader-progress-4"
        mode="query"
        role="progressbar"
      >
        <div
          className="MuiLinearProgress-bar-25 MuiLinearProgress-primaryColorBar-20 MuiLinearProgress-indeterminateBar1-30"
          style={Object {}}
        />
        <div
          className="MuiLinearProgress-bar-25 MuiLinearProgress-primaryColorBar-20 MuiLinearProgress-indeterminateBar2-31"
          style={Object {}}
        />
      </div>
    </td>
  </tr>
</table>
`;
